widget/embedded/src/components/SwapDetails/SwapDetails.tsx

```diff
@@ -90,12 +92,27 @@ export function SwapDetails(props: SwapDetailsProps) {
     setModalState(null);
   };
 
+  const getUnreadNotifications =
+    useNotificationStore.use.getUnreadNotifications();
+  const setAsRead = useNotificationStore.use.setAsRead();
+  const unreadNotifications = getUnreadNotifications();
   const currentStep = getCurrentStep(swap);
   const currentStepNetworkStatus = currentStep?.networkStatus;
 
   useEffect(() => {
-    if (swap.status === 'success' || swap.status === 'failed') {
-      setShowCompletedModal(swap.status);
+    const existNotification = unreadNotifications.find(
+      (n) => n.requestId === swap.requestId
+    );
+    if (existNotification) {
+      const {
+        event: { type },
+      } = existNotification;
+      const isUnread =
```

Comment:
Is there any reason that you check unread notification by this condition instead of reading from store?

Reply:
We don't need it. I removed it.